tex/generic/pgf/frontendlayer/tikz/tikz.code.tex

```diff
@@ -2580,28 +2580,64 @@
 
 \def\tikz@fchar oreach{\tikz@foreach}%
 
-\def\tikz@foreach{%
+\def\tikzforeach@smugglers@cove{}
+\let\tikztostart=\relax
+
+\def\tikzforeach@smuggle@macro#1{%
+  \pgfutil@ifx#1\relax{}{%
+    \expandafter
+    \pgfutil@g@addto@macro
+    \expandafter
+    \tikzforeach@smugglers@cove
+    \expandafter
+    {%
+      \expandafter\def\expandafter#1\expandafter{#1}%
+    }%
+  }%
+}
+
+\def\tikzforeach@smuggle@dimen#1{%
+  \pgfutil@ifx#1\relax{}{%
+    \expandafter
+    \pgfutil@g@addto@macro
+    \expandafter
+    \tikzforeach@smugglers@cove
+    \expandafter
+    {%
+      \expandafter#1\expandafter=\the#1\relax%
+    }%
+  }%
+}
+
+\def\tikzforeach@smugglers@loot{%
+  %
+  \tikzforeach@smuggle@macro\tikz@moveto@waiting%
+  \tikzforeach@smuggle@macro\tikztostart%
+  \tikzforeach@smuggle@macro\tikz@tangent%
+  %
+  \tikzforeach@smuggle@dimen\tikz@lastx%
+  \tikzforeach@smuggle@dimen\tikz@lasty%
+  \tikzforeach@smuggle@dimen\tikz@lastxsaved%
+  \tikzforeach@smuggle@dimen\tikz@lastysaved%
+  %
+}
```

Comment:
I like this approach (and the naming scheme :grin:), but I think it would be even better if we could make this more general using `pgfkeys`, so that a user can register their own loot.

Reply:
Are you happy with just `remember macro` and `remember dimension` for the keys?  I've not found any other keys in the tikz namespace with those names.